_includes/sidebar.html

```diff
@@ -52,6 +52,15 @@
   </ul> <!-- ul.nav.flex-column -->
 
   <div class="sidebar-bottom mt-auto d-flex flex-wrap justify-content-center align-items-center">
+    {% unless site.theme_mode %}
+      {% if site.data.contact.size > 0 %}
+        <span class="icon-border order-2"></span>
+      {% endif %}
+
+      <span id="mode-toggle-wrapper" class="order-1" tabindex="0">
+        {% include mode-toggle.html %}
+      </span>
```

Comment:
Please restore the changes of `_includes/sidebar.html`

Reply:
> tab order for darkmode should come before other social media link...

In the design expectation, this is to make the mode-switching JS code not block the rendering of social icons, which is why `.order-x` is specified.

> The best option would be to use button which would save us from the hassle of using custom event listeners...

Can you please explain why the button is better than EventListener in this case?

Reply:
- How about if instead of inlining js we use <script async/defer src="mode-toggle"> javascript. In this way rendering of social icons won't be blocked.
- Note that I have added one attribute tabindex="0", to have the correct tab order, in case you missed.
- checkout this for button soln, https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_button_test, although only onclick was specified but when you tab focus it and press enter it still works.

Reply:
Sorry, I missed the `tabindex=0`, you can keep it, but please restore the code block to the bottom. My goal is to execute the mode detection JS immediately. If we use `async/defer`, when the session has a theme-mode state and the user jumps to another page, there will be a dark/light switch flicker, so please be sure to keep the original JS running priority.

In addition, the button is just another method of implementation, and there is no outstanding advantage to replacing the event listener.

Reply:
Agreed with your first point.

Regarding advantage of using button element, for the people in accessibility need when they use screen reader, it explicitly says that this is a "button" which is not the cases with using span element. Also, for the programmer it is more intuitive when he/she is going through the code. Yes at the end result will be same visually but again not all people consume/interact with the web content visually. I hope you get my point. Although not making any further changes for now. Thanks and have a nice day :)

Reply:
> for the programmer it is more intuitive when he/she is going through the code...

True, I agree with this 👍